### PR TITLE
Homework Podstawy C++ password-check

### PR DESCRIPTION
**Pull request by Martyna-Tu**
@Martyna-Tu, @michalsoloducha

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @michalsoloducha

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @Martyna-Tu

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @michalsoloducha

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @Martyna-Tu

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @michalsoloducha

**Comment by ziobron** [comment]
⏱ 3 XP extra for @michalsoloducha for on-time delivery

File: homework/password-check/.bonus.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <algorithm>
+#include <cctype>
+#include <string>
+
+std::vector<ErrorCode> checkedPasswordRules(std::string str){
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/.bonus.cpp:7]: Redundant new line at the beginning of function.

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+
+    if(std::none_of(str.cbegin(), str.cend(), [](unsigned char c){ return std::isdigit(c);})){
+        return static_cast<ErrorCode>(2);
+    }
+    
+    if(std::all_of(str.cbegin(), str.cend(), [](unsigned char c){ return std::isalnum(c);})){
+        return static_cast<ErrorCode>(3);
+    }
+
+    if(std::none_of(str.cbegin(), str.cend(), [](unsigned char c){ return std::isupper(c);})){
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:36]: Add space after `if`, `for`, `while` `function` etc...

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**